test-case/test-jack-detection-playback-capture.sh

```diff
@@ -0,0 +1,206 @@
+
+func_opt_parse_option "$@"
+
+tplg=${OPT_VAL['t']}
+relay=${OPT_VAL['u']}
+loop_cnt=${OPT_VAL['l']}
+
+TWO_SECONDS=2
+
+check_control_switch_state()
+{
+    # Check the state of the switch using amixer.
+    # The switch name is passed as the first argument, and the expected state (on/off)
+    # is passed as the second argument.
+    # Returns 0 if the state matches, 1 otherwise.
+    local control_name="$1"
+    local expected_control_state="$2"
+    local control_state
+
+    control_state=$(amixer -c "$SOFCARD" contents | awk -v name="$control_name" '
+        BEGIN {
+            RS = "";
+            IGNORECASE = 1;
+            split(name, parts, " ");
+        };
+        $0 ~ parts[1] && $0 ~ parts[2] {
+            if (match($0, /values=(on|off)/, m)) print m[1];
+        }
+    ')
+    dlogi "$control_name switch is: $control_state"
+
+    if [[ "$expected_control_state" == "$control_state" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+testing_one_pcm()
+{
+    dlogi "===== Testing: (PCM: $pcm [$dev]<$type>) (Loop: $i/$loop_cnt) ====="
+    dlogi "DEVICE: $dev, TYPE: $type, PCM: $pcm, RATE: $rate, CHANNEL: $channel"
+
+    dlogi "Command: aplay -Dplug$dev -q /dev/zero"
+    # Start alsabat in background with longer duration
+    dlogi "Starting aplay in background..."
+    aplay "-Dplug$dev" -d10 -q /dev/zero & pid_playback=$! || {
+        func_lib_lsof_error_dump "$snd"
+        die "Failed to start aplay on PCM: $pcm"
+    }
+    # Wait briefly to ensure the device is ready and avoid read errors
+    sleep 1
+    dlogi "aplay started with PID: $pid_playback"
+
+    dlogi "Unplug jack audio."
+    usbrelay_switch "$relay" 1
+
+    # Wait for a short period to allow the system to detect the unplug event
+    sleep $TWO_SECONDS
+
+    # check if the aplay process is still running after unplugging the jack
+    ps -p "$pid_playback" > /dev/null || {
+        func_lib_lsof_error_dump "$snd"
+        die "Playback process terminated unexpectedly after unplugging the jack."
+    }
+
+    check_control_switch_state "headset" "off" || {
+        die "unplug headset jack failed."
+    }
+
+    check_control_switch_state "headphone" 'off' || {
+        die "unplug headphone jack failed."
+    }
+
+    dlogi "Plug jack audio."
+    usbrelay_switch "$relay" 0
+
+    # Wait for a short period to allow the system to detect the plug event
+    sleep $TWO_SECONDS
+
+    # check if the aplay process is still running after unplugging the jack
+    ps -p "$pid_playback" > /dev/null || {
+        func_lib_lsof_error_dump "$snd"
+        die "Playback process terminated unexpectedly after unplugging the jack."
+    }
+
+    check_control_switch_state "headset" "on" || {
+        die "Plug headset jack failed."
+    }
+
+    check_control_switch_state "headphone" "on" || {
+        die "Plug headphone jack failed."
+    }
+
+    kill -9 $pid_playback > /dev/null 2>&1
+    wait $pid_playback 2>/dev/null || true
```

Comment:
I hope we already have some code for this, maybe in `case-lib/`

(every script tries to re-invent this)